tests/director/tests/formation_api_test.go

```diff
@@ -1095,6 +1095,281 @@ func TestFormationNotifications(stdT *testing.T) {
+
+	certSecuredHTTPClient := &http.Client{
+		Timeout: 10 * time.Second,
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				Certificates: []tls.Certificate{
+					{
+						Certificate: cc.Get().Certificate,
+						PrivateKey:  cc.Get().PrivateKey,
+					},
+				},
+				ClientAuth:         tls.RequireAndVerifyClientCert,
+				InsecureSkipVerify: conf.SkipSSLValidation,
+			},
+		},
+	}
+
+	defer cleanupNotificationsFromExternalSvcMock(t, certSecuredHTTPClient)
+
+	body := getNotificationsFromExternalSvcMock(t, certSecuredHTTPClient)
+	assertNotificationsCountForTenant(t, body, app1.ID, 1)
+
+	notificationsForApp1 := gjson.GetBytes(body, app1.ID)
+	assignNotificationAboutApp2 := notificationsForApp1.Array()[0]
+	assertFormationNotification(t, assignNotificationAboutApp2, "assign", formation.ID, app2.ID, localTenantID2, appNamespace, appRegion)
+
+	t.Logf("Unassign Application 1 from formation %s", formationName)
+	unassignReq = fixtures.FixUnassignFormationRequest(app1.ID, "APPLICATION", formationName)
+	err = testctx.Tc.RunOperationWithCustomTenant(ctx, certSecuredGraphQLClient, tnt, unassignReq, &unassignFormation)
+	require.NoError(t, err)
+	require.Equal(t, formationName, unassignFormation.Name)
+
+	body = getNotificationsFromExternalSvcMock(t, certSecuredHTTPClient)
+	assertNotificationsCountForTenant(t, body, app1.ID, 2)
+
+	notificationsForApp1 = gjson.GetBytes(body, app1.ID)
+	unassignNotificationFound := false
+	for _, notification := range notificationsForApp1.Array() {
+		op := notification.Get("Operation").String()
+		if op == "unassign" {
+			unassignNotificationFound = true
+			assertFormationNotification(t, notification, "unassign", formation.ID, app2.ID, localTenantID2, appNamespace, appRegion)
+		}
+	}
+	require.True(t, unassignNotificationFound, "notification for unassign app2 not found")
+
+	t.Logf("Assign application 1 to formation %s again", formationName)
+	assignReq = fixtures.FixAssignFormationRequest(app1.ID, "APPLICATION", formationName)
+	err = testctx.Tc.RunOperationWithCustomTenant(ctx, certSecuredGraphQLClient, tnt, assignReq, &assignedFormation)
+	require.NoError(t, err)
+	require.Equal(t, formationName, assignedFormation.Name)
+
+	body = getNotificationsFromExternalSvcMock(t, certSecuredHTTPClient)
+	assertNotificationsCountForTenant(t, body, app1.ID, 3)
+
+	notificationsForApp1 = gjson.GetBytes(body, app1.ID)
+	assignNotificationsFound := 0
+	for _, notification := range notificationsForApp1.Array() {
+		op := notification.Get("Operation").String()
+		if op == "assign" {
+			assignNotificationsFound++
+			assertFormationNotification(t, notification, "assign", formation.ID, app2.ID, localTenantID2, appNamespace, appRegion)
+		}
+	}
+	require.Equal(t, 2, assignNotificationsFound, "two notifications for assign app2 expected")
+
+	t.Logf("Unassign Application 2 from formation %s", formationName)
+	unassignReq = fixtures.FixUnassignFormationRequest(app2.ID, "APPLICATION", formationName)
```

Comment:
maybe use `string(graphql.FormationObjectTypeApplication)` here and wherever it's needed